### PR TITLE
Raise `UpstreamThumbnailException` for unsuccessful requests

### LINKED ISSUE #676: UpstreamThumbnailException only raised when there are "ambiguous" problems and not for things outside the 2xx range
## Description

Requests does not raise an exception if a link 404s, you have to check the status code or call `raise_for_status` on the response.

## Reproduction

1.  See the response from `https://api-dev.openverse.engineering/v1/images/aaabb3e4-cbd9-4a55-b16f-634fd9d91483/thumb/` which should be 424ing is instead 404ing (check the actual status code of the response, not the response contents which currently relays the upstream issue).

## Additional context
Noticed while testing the [v2.5.11 release](https://github.com/WordPress/openverse-api/releases/tag/v2.5.11) but did not stop the release because we're still no longer 500'ing.

## Resolution

- [ ] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by krysal**
## Fixes

Fixes #676 by @sarayourfriend

## Description

This PR extracts the params building from the `photon.get` method to reduce it a bit, and also calls `raise_for_status` from the response, as suggested in the issue, to make sure we raise the correct custom exception from unsuccessful HTTP response codes.

The `requests.HTTPError` class inherits from `requests.RequestException`[^1] so there is no need to add it explicitly to the `except` block.

[^1]: https://docs.python-requests.org/en/latest/user/quickstart/#errors-and-exceptions

## Testing Instructions

1. To test this manually, try replacing one DB entry's URL with the sample image URL provided in the issue:
```
https://upload.wikimedia.org/wikipedia/commons/8/89/Flag_of_the_Turkic_Council.svg
```
2. Make sure that image doesn't have a thumbnail URL; in that case, make it null.
3. Go to the image's thumbnail URL (`http://localhost:50280/v1/images/<uuid>/thumb/`) and make sure you see a response like the following:
```
HTTP 424 Failed Dependency
Allow: GET, HEAD, OPTIONS
Content-Type: application/json
Vary: Accept

{
    "detail": "Failed to render thumbnail. 404 Client Error: File Not Found for url: https://i0.wp.com/upload.wikimedia.org/wikipedia/commons/8/89/Flag_of_the_Turkic_Council.svg?w=600&quality=80&ssl=true"
}
```

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [N/A] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: api/catalog/api/utils/photon.py [hunk 1]
```diff
@@ -58,35 +50,40 @@ def get(
...
+
+def get(
+    image_url: str,
+    accept_header: str = "image/*",
+    is_full_size: bool = False,
+    is_compressed: bool = True,
+) -> HttpResponse:
+    logger = parent_logger.getChild("get")
+    # Photon options documented here:
+    # https://developer.wordpress.com/docs/photon/api/
```
Comment (sarayourfriend) [inline api/catalog/api/utils/photon.py:64]: nit: this comment should probably be in the `_get_photon_params` function instead.

File: api/test/unit/utils/photon_test.py [hunk 2]
```diff
@@ -287,3 +284,15 @@ def test_get_successful_https_image_url_sends_ssl_parameter(mock_image_data):
...
+@pook.on
+def test_get_unsuccessful_request_raises_exception():
+    mock_get: pook.Mock = pook.get(PHOTON_URL_FOR_TEST_IMAGE).reply(404).mock
+
+    with pytest.raises(
+        UpstreamThumbnailException, match=r"Failed to render thumbnail."
+    ):
+        photon_get(TEST_IMAGE_URL)
+
+    assert mock_get.matched
```
Comment (sarayourfriend) [inline api/test/unit/utils/photon_test.py:298]: Isn't this test semi-redundant with the one that uses mocking to force requests to throw an error? I suppose it's more realistic to do it this way (this feels less implementation-detail oriented) but consolidating the two would be helpful for keeping the test purposes clear.

**Review by sarayourfriend — APPROVED** [review]
LGTM. Small non-blocking request for test consolidation but otherwise the changes look like a great improvement.

**Review by obulat — APPROVED** [review]
LGTM!

**Pull request merged**